pycalphad/model.py

```diff
@@ -463,13 +451,18 @@ def _array_validity(self, constituent_array):
             # species can be specified in the anion sublattice without any
             # species in the cation sublattice.
             if self._dbe.phases[self.phase_name].model_hints.get('ionic_liquid_2SL', False):
-                pass
+                for sublattice in constituent_array:
+                    if (set(sublattice).issubset(self.constituents[1]) \
```

Comment:
Is this `self.constituents[1]` appropriate here, or would be better to check if parameter sublattice is contained in `self.components`?

I think this PR is ready or almost ready. Do you think that should I remove the excessive tests and keep just the test now named as `test_model_energy`?